### ISSUE TITLE
Configurable scroll padding

### ISSUE THREAD

**Issue by kurmachu** [microsoft/vscode#131203]
I'd like to be able to configure an amount of characters (or lines) of padding for both horizontal and vertical scrolling (using the text cursor).
Currently, vscode has 0 lines of vertical padding, and ~2 characters of padding for horizontal scrolling. I'd like to be able to configure this in settings, to show more or less while scrolling.

Currently:
![image](https://user-images.githubusercontent.com/31492625/130136243-2e72890a-9846-4a13-85f5-7d80efedfaa6.png)

Example of modified padding:
![image](https://user-images.githubusercontent.com/31492625/130136501-e4e96cd8-e1f7-422b-ad22-95860e69d8ad.png)

**Comment by alexdima** [comment]
🤖 Thanks for the suggestion! This is actually already possible with existing settings:

- **`editor.cursorSurroundingLines`** — controls the minimum number of visible lines above and below the cursor (vertical padding). Known as `scrollOff` in Vim.
- **`editor.scrollBeyondLastColumn`** — controls extra horizontal scroll padding.

These have been available since 2019 and should cover the use case described here. Closing as the functionality is already provided.

**Comment by kurmachu** [comment]
`editor.cursorSurroundingLines` (or `editor.stickyScroll.maxLineCount`, whichever is greater) does in fact cover the vertical aspect of this issue, which does make it embarrassing that I entirely missed that when making this issue. However, this feature request also has a horizontal component.

`editor.scrollBeyondLastColumn` does _not_ "control extra horizontal scroll padding" for the cursor, it controls the amount you can scroll past the end of a row of text with the mouse or scrollbar (it's in the name; scroll beyond last column). Because it is explicitly _not_ padding on cursor movement scroll, it is actually impossible to even see this space when scrolling entirely with cursor movements.

Here is a demonstration with `"editor.scrollBeyondLastColumn": 10,`, where I first move the cursor to the right with arrow keys, then jump to the end of the line with the end key, and then finally scroll the `scrollBeyondLastColumn` gap by using a mouse. 

https://github.com/user-attachments/assets/3ea71b4a-a1f9-4deb-8732-7e3bb525fcb3

You can see that the cursor's distance from the scrollbar is always ~2 characters, despite the setting.

I wasn't able to find an `editor.cursorSurroundingLines` equivalent for horizontal scrolling after (hopefully) checking every search result for "scrolling" in the editor and trying to look online (though then again I also wasn't able to find `editor.cursorSurroundingLines` itself originally).

**Comment by alexdima** [comment]
You’re right, `editor.scrollBeyondLastColumn` only affects the scrollable width and does not control cursor reveal padding. Cursor movement uses separate, currently non-configurable horizontal padding. Reopening and narrowing this issue to configurable horizontal cursor-surrounding space. Thanks for the correction.
